backend/src/akvo/lumen/component/keycloak.clj

```diff
@@ -0,0 +1,139 @@
+  (users [this tenant-label] "List tenants users")
+  (add-user-with-email [this tenant-label email] "Add user to tenant")
+  (user? [this email] "Predicate to see if the email has a user in KC"))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Helpers
+;;;
+
+(defn fetch-openid-configuration
+  "Get the openid configuration"
+  [issuer]
+  (-> (client/get (format "%s/.well-known/openid-configuration" issuer))
+      :body json/decode))
+
+(defn request-options
+  [{:keys [openid-config credentials]}]
+  (let [params (merge {"grant_type" "client_credentials"}
+                      credentials)
+        resp (client/post (get openid-config "token_endpoint")
+                          {:form-params params})
+        access-token (-> resp :body json/decode (get "access_token"))]
+    {:headers {"Authorization" (str "bearer " access-token)
+               "Content-Type" "application/json"}}))
+
+(defn group-by-path
+  [{:keys [api-root credentials openid-config]} path request-options]
+  (-> (client/get (format "%s/group-by-path/%s"
+                          api-root path)
+                  request-options)
+      :body json/decode))
+
+(defn group-members
+  [{:keys [api-root credentials openid-config]} group-id request-options]
+  (-> (client/get (format "%s/groups/%s/members"
+                          api-root group-id)
+                  request-options)
+      :body json/decode))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Helpers
+;;;
+
+;;; This is kinda fragile, this relies on that one user not belog to multiple
+;;; groups in a tenant and does not support new sub groups!!!!!!
+;;; Hence making someone an admin would require us to add the user to t1/admin
+;;; and also remove the user from the t1 group
+(defn tenant-members
+  "Return the users for a tenant. The tenant label here becomes the group-name"
+  [keycloak tenant-label]
+  (try
+    (let [options (request-options keycloak)
+          tenant-group (group-by-path keycloak tenant-label options)
+          group-id (get tenant-group "id")
+          users (group-members keycloak group-id options)
+          admin-group-id (get (first (filter #(= "admin"
+                                                 (get % "name"))
+                                             (get-in tenant-group ["subGroups"])))
+                              "id")
+          admins (map #(assoc % "admin" true)
+                      (group-members keycloak admin-group-id options))
+          members (filter #(and (get % "emailVerified") (get % "enabled"))
+                          (concat admins users))
+          response-filter ["admin" "email" "firstName" "id" "lastName"
+                           "username"]]
+      (response (map #(select-keys % response-filter) members)))
+    (catch Exception e
+      (prn e)
+      (let [ed (ex-data e)]
+        (prn ed)
+        (response {:status (:status ed)
```

Comment:
You catch `Exception` but assume it's an `ExceptionInfo` thrown from the http client. If it's another kind of exception `ex-data` will return null, and so will `(:status ed)`.